cluster/examples/kubernetes/ceph/create-external-cluster-resources.py

```diff
@@ -395,15 +392,20 @@ def _gen_output_map(self):
         self.out_map['ROOK_EXTERNAL_USER_SECRET'] = self.create_checkerKey()
         self.out_map['CSI_RBD_NODE_SECRET_SECRET'] = self.create_cephCSIKeyring_RBDNode()
         self.out_map['CSI_RBD_PROVISIONER_SECRET'] = self.create_cephCSIKeyring_RBDProvisioner()
-        self.out_map['CSI_CEPHFS_NODE_SECRET'] = self.create_cephCSIKeyring_cephFSNode()
-        self.out_map['CSI_CEPHFS_PROVISIONER_SECRET'] = self.create_cephCSIKeyring_cephFSProvisioner()
+        self.out_map['CEPHFS_POOL_NAME'] = self._arg_parser.cephfs_data_pool_name
+        self.out_map['CEPHFS_FS_NAME'] = self._arg_parser.cephfs_filesystem_name
+        self.out_map['CSI_CEPHFS_NODE_SECRET'] = ''
+        self.out_map['CSI_CEPHFS_PROVISIONER_SECRET'] = ''
+        # we don't have the MDS,
+        # make sure we don't set CephFS node and provisioner keyring
+        if self.out_map['CEPHFS_FS_NAME'] and self.out_map['CEPHFS_POOL_NAME']:
```

Comment:
why cephfs is special here why not for RBD?

Reply:
RBD will always be present since the cluster always has OSDs running, but not necessarily MDSs, that's why it's optional.